controller/test/pid_lib/pid_test.cpp

```diff
@@ -0,0 +1,202 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ * module contributors: asmodai27
+ *
+ * The purpose of this module is to provide unit testing for the pid
+ * controller module.  Module is to be run on an x86 host and is not to be run
+ * on an Arduino platform.
+ */
+
+#include "hal.h"
+#include "pid.h"
+#include "types.h"
+#include "gtest/gtest.h"
+
+//@TODO: Adjust this tolerance... assumes the PWM is a 0-255 integer but that
+// may not be the case
+static const double output_TOLERANCE = 1;
+
+// PID
+static double setpoint;
+static double input;
+static double output;
+
+// default PID gains
+static float Kp;
+static float Ki;
+static float Kd;
+
+// PID Sample time (in milliseconds)
+static uint32_t sample_time = 100;
+
+// Maximum task jitter (assume 5 ms for now, will need adjuting)
+static const uint32_t max_task_jitter = 5;
+
+TEST(pidTest, Proportional) {
+  Kp = 0.9;
+  Ki = 0;
+  Kd = 0;
+  setpoint = 25;
+  input = 10;
+  PID myPID(&input, &output, &setpoint, Kp, Ki, Kd, DIRECT);
```

Comment:
Based on the usage here, I think Kp Ki Kd don't have to be static variables, they can be local variables, right?

Reply:
indeed, I planned on using the same gains in all tests but then changed my plans, will update.